packages/components/src/color-picker/test/index.tsx

```diff
@@ -342,4 +345,89 @@ describe( 'ColorPicker', () => {
 			} );
 		} );
 	} );
+
+	describe.each( [
+		[ 'hsl', 'HSL' ],
+		[ 'rgb', 'RGB' ],
+	] )( 'Alpha-enabled %s format', ( format, formatLabel ) => {
+		it( `should update alpha correctly when ${ formatLabel } format is selected`, async () => {
+			const user = userEvent.setup();
+			const onChange = jest.fn();
+
+			render(
+				<ControlledColorPicker
+					onChange={ onChange }
+					enableAlpha
+					initialColor="#ffffff80"
+				/>
+			);
+
+			const formatSelector = screen.getByRole( 'combobox' );
+			expect( formatSelector ).toBeVisible();
+			await user.selectOptions( formatSelector, format );
+
+			const alphaInput = screen.getByRole( 'spinbutton', {
+				name: 'Alpha',
+			} );
+			expect( alphaInput ).toBeVisible();
+
+			const alphaSliders = screen.getAllByRole( 'slider', {
+				name: 'Alpha',
+			} );
+
+			expect( alphaSliders ).toHaveLength( 2 );
+
+			// Choose the second slider which is the actual slider of type: input[type="range"]
+			const alphaSlider = alphaSliders.at( -1 )!;
+
+			expect( alphaSlider ).toHaveValue( '50' );
+			expect( alphaInput ).toHaveValue( 50 );
+
+			expect( onChange ).not.toHaveBeenCalledWith();
+
+			// Test pattern 1: Update the slider
+			fireEvent.change( alphaSlider, {
+				target: { value: 75 },
+			} );
+
+			await waitFor( () => {
+				expect( onChange ).toHaveBeenCalledTimes( 1 );
+			} );
+			await waitFor( () => {
+				expect( onChange ).toHaveBeenCalledWith( '#ffffffbf' );
+			} );
+
+			await waitFor( () => {
+				expect( alphaInput ).toHaveValue( 75 );
+			} );
+			await waitFor( () => {
+				expect( alphaSlider ).toHaveValue( '75' );
+			} );
```

Comment:
We probably don't need to wrap these 3 assertions in `await waitFor`. Once we `await` for `onChange` to have been called, the rest of the assertions can already be checked without the need for waiting:

```suggestion
			expect( onChange ).toHaveBeenCalledWith( '#ffffffbf' );
			
			expect( alphaInput ).toHaveValue( 75 );

			expect( alphaSlider ).toHaveValue( '75' );
```